nanopub/nanopub.py

```diff
@@ -200,114 +201,117 @@ class NanopubClient:
         headers = {"Accept": "application/json"}
-        r = requests.get(apiurl, params=searchparams, headers=headers)
+        r = None
+        for grlc_url in self.grlc_urls:
+            url = grlc_url + endpoint
+            r = requests.get(url, params=params, headers=headers)
+            if r.status_code == 502:  # Server is likely down
+                warnings.warn(f'Could not get response from {grlc_url}, trying other servers')
+            else:
+                r.raise_for_status()  # For non-502 errors we don't want to try other servers
+                return r
 
-        if r.ok:
+        raise requests.HTTPError(f'Could not get response from any of the nanopub grlc '
+                                 f'endpoints, last response: {r.status_code}:{r.reason}')
 
-            # Make sure that results are provided
-            try:
-                results_json = r.json()
-            except:
-                # If the returned message can't be serialized as JSON (such as due to virtuoso error) then there are no results
-                print('Error: Could not serialize response as JSON:\n', r.content)
-                return []
+    def _search(self, endpoint, params, max_num_results):
+        """
+        General nanopub server search method. User should use e.g. search_text() or search_pattern() instead.
 
-            results_list = results_json['results']['bindings']
-            nanopubs = []
+        Raises:
+            JSONDecodeError: in case response can't be serialized as JSON, this can happen due to a
+                virtuoso error.
+        """
+        r = self._query_grlc(params, endpoint)
+        results_json = r.json()
 
-            for result in results_list:
-                nanopub = {}
-                nanopub['np'] = result['np']['value']
+        results_list = results_json['results']['bindings']
+        nanopubs = []
 
-                if 'v' in result:
-                    nanopub['description'] = result['v']['value']
-                elif 'description' in result:
-                    nanopub['description'] = result['description']['value']
-                else:
-                    nanopub['v'] = ''
+        for result in results_list:
+            nanopub = {}
+            nanopub['np'] = result['np']['value']
 
-                nanopub['date'] = result['date']['value']
+            if 'v' in result:
+                nanopub['description'] = result['v']['value']
+            elif 'description' in result:
+                nanopub['description'] = result['description']['value']
+            else:
+                nanopub['v'] = ''
 
-                nanopubs.append(nanopub)
+            nanopub['date'] = result['date']['value']
 
-                if len(nanopubs) >= max_num_results:
-                    break
+            nanopubs.append(nanopub)
 
-            return nanopubs
+            if len(nanopubs) >= max_num_results:
+                break
```

Comment:
I didn't change anything here, github is just confused.

Reply:
Ha, I'll trust you... this time... :)